src/api/type-defs/objects/PageInfo.ts

```diff
@@ -5,6 +5,12 @@ export default /* GraphQL */ `
     next: String
     hasNext: Boolean
   }
+
+  type PageCount {
+    count: Int
+  }
+
+  union PageInfoWithCount = PageInfo | PageCount
```

Comment:
@alukach I may be misunderstanding GraphQL unions, but my understanding is that the `union PageInfoWithCount = PageInfo | PageCount` means that it can be either `PageInfo` or `PageCount` types, but I think what we want is a way to extend the base `PageInfo` properties to have an additional `count` property for use in `ImageErrorsConnection`. 
 
If you check out the [return payload](https://github.com/tnc-ca-geo/animl-api/blob/eead714ee93d7cb2b5e6372e72dd024f3af24c9c/src/api/resolvers/Query.js#L78) from the `imageErrors` resolver, it includes all of the `PageInfo` fields plus an additional `count` field.

Reply:
Also it seems to be failing the tests.

Reply:
Good catch, union doesn't mean extends, my mistake. I've updated this in [9caf1e5](https://github.com/tnc-ca-geo/animl-api/pull/197/commits/9caf1e5c5091f681b487fbc2d99eb3d242f61fe5)

Reply:
I just did some more reading and I have to say I don't care for this one bit, but apparently GraphQL [doesn't support the ability to simply extend a type](https://stackoverflow.com/questions/48917863/how-do-you-extend-types-in-graphql) and call it something else. The GraphQL concept of `interfaces` is also kind of lame IMO, but I think we need to tweak what you have to look like this to properly use and `implement` an interface:

```javascript
export default /* GraphQL */ `
  // the shared properties are the "interface"
  interface Pagination {
    previous: String
    hasPrevious: Boolean
    next: String
    hasNext: Boolean
  }

  // still need to write out all the shared interface props again for some reason
  type PageInfo implements Pagination {
    previous: String
    hasPrevious: Boolean
    next: String
    hasNext: Boolean
  }

  // this one looks good
  type PageInfoWithCount implements Pagination {
    previous: String
    hasPrevious: Boolean
    next: String
    hasNext: Boolean
    count: Int
  }
`;
```

I don't really understand what the point of `interfaces` are other than to yell at you if you add a new prop to the interface but forget to add it to the actual types that implement them. But alas. Here are the docs I was referencing: https://graphql.com/learn/interfaces-and-unions/